sql/sql_parse.h

```diff
@@ -172,13 +172,23 @@ extern uint sql_command_flags[];
       wsrep_to_isolation_begin(thd, db_, table_, table_list_))                \
     goto wsrep_error_label;
 
-#define WSREP_TO_ISOLATION_BEGIN_FK_TABLES_IF(db_, table_, table_list_,       \
+/* Checks if lex->no_write_to_binlog is set for statements that use
+  LOCAL or NO_WRITE_TO_BINLOG
+*/
+#define WSREP_TO_ISOLATION_BEGIN_FK_TABLES_IF_WRTCHK(db_, table_, table_list_,       \
```

Comment:
Why the code duplication for the condition, it could reuse the other macro?

Also, shouldn't the formatting of the lines below change?

Reply:
`IF_WRTCHK` versions add `thd->lex->no_write_to_binlog` condition in `if`. This condition has to be checked after `WSREP(thd)` which validates `thd` pointer, so no easy/clean way to reuse `WSREP_TO_ISOLATION_BEGIN_FK_TABLES_IF` in `WSREP_TO_ISOLATION_BEGIN_FK_TABLES_IF_WRTCHK'. 
That would need introducing some intermediate macros making things more obfuscated.

Another way of doing it would be doing something like:
```
#define WSREP_TO_ISOLATION_BEGIN_FK_TABLES_IF_WRTCHK if (thd != nullptr && !thd->lex->no_write_to_binlog && WSREP(thd) ...
``
This way `thd` would be validated against nullptr twice, which should be optimized by the compiler.
Additionally, I see `WSREP_TO_ISOLATION_BEGIN_(WRTCHK)` has the same code duplication and exactly the same reasoning as above.
Let's leave it as it is.